packages/components/transitions/src/collapse.tsx

```diff
@@ -113,9 +112,9 @@ export const Collapse = forwardRef<CollapseProps, "div", false>(
       if (isBrowser) setMounted(true)
     }, [])
 
-    const animate = isOpen || unmountOnExit ? "enter" : "exit"
+    const animate = rest.isOpen || unmountOnExit ? "enter" : "exit"
 
-    isOpen = unmountOnExit ? isOpen : true
+    const isOpen = unmountOnExit ? rest.isOpen : true
```

Comment:
Why was it necessary to modify this section?If you don't need to fix it, please revert back to the original.

Also, this modification will cause a warning on the console because of the `isOpen` in `rest`.